### DISCUSSION TITLE
what does vercel actually expose to public for my sveltekit deployment?

### DISCUSSION THREAD

**Discussion by gabrielchl** [vercel/community#1780] [Help]
### Summary

so I've created a sveltekit application
there's 1 file in there that must be kept away from the front end: `src/routes/verify-oauth-code/+server.ts` as it includes my oauth client secret (from env vars using `$env/static/private`)

to dpeloy to vercel i simply selected the repository in vercel and used the default options, the vercel adapter for svelte kit wasn't used

however, upon deployment to vercel, usage shows that serverless functions nor edge functions were being used
i'm worried that it's somehow serving the whole application (including the sensitive bits) to the FE, as I have no idea what the default settings for vercel are / what's actually deployed / exposed to the FE

is that's what's happening? do i need to add some config to explicitly tell vercel to serve the server files in a server instead?

**Comment by amyegan** [comment]
Hi Gabriel. Are you using an older version of SvelteKit? If you don't have a specific version pinned, then you may need to change `+server.ts` to [`+page.server.ts`](https://kit.svelte.dev/docs/load#:~:text=If%20your,on%20the%20server). It's also a good idea to double check that you are either using [adapter-auto](https://kit.svelte.dev/docs/adapter-auto) for zero config deployment or [@sveltejs/adapter-vercel](https://kit.svelte.dev/docs/adapter-vercel)

**Reply by gabrielchl** [reply]
hey, i'm using 1.5.0 of sveltekit and am using @sveltejs/adapter-auto in svelte.config.js
the file in question is the highlighted verify-oauth-code/+server.ts file, it's +server.ts instead of +page.server.ts because it's contains a `GET` function but no the `load` ones
i think it's [this](https://kit.svelte.dev/docs/routing#server) in sveltekit docs and [this](https://vercel.com/docs/beginner-sveltekit/endpoints) in vercel docs
<img width="300" alt="Screenshot 2023-03-14 at 15 18 45" src="https://user-images.githubusercontent.com/5616556/225048538-13d59e08-2761-456f-a773-ba49833e461d.png">

**Reply by gabrielchl** [reply]
i guess my question really is: what's the default behaviour of deploying a sveltekit app with all the default configs, especially in the context of the server endpoints? (recent version of sveltekit, using adapter-auto)

**Reply by amyegan** [reply]
I see what you're asking now. Sorry for my misunderstanding. I don't know the exact mechanics, but I believe all `+server.js` and `+page.server.js` server-side code is bundled to run in a serverless function. I'm double checking with the Svelte team to see if there are any scenarios in which server-side code would somehow make it to the client

**Reply by gabrielchl** [reply]
np! yeah that's what i expected to happen too, however my dashboard is showing 0 serverless functions usage thus the concern, i'll look into it on my end as well :)
"worst"-case scenario i switch to the vercel adaptor and explicitly state that the file is meant for serverless.

**Reply by amyegan** [reply]
I confirmed with the Svelte team that everything is bundled into a single serverless function by default.

> *server.js (or .ts, etc) files cannot end up in the client-side bundle — the app will simply fail to build if you try. the only way you could get sensitive stuff into the client is by not putting it in a *server.js file, or by explicitly returning it from a load function or a request handler

You can view the functions from the deployment details page, but it will look like one big serverless function for SvelteKit. If you are able to access the deployed API route, then it's working as expected. Otherwise, explicitly setting the Vercel adapter instead of letting it auto-select is a good next step

**Reply by gabrielchl** [reply]
ah cool!
one last question, looking into the build logs, it does indeed seem like everything was bundled into one serverless function
```
Build Completed in /vercel/output [1m]
--
22:23:36.522 | Generated build outputs:
22:23:36.522 | - Static files: 27
22:23:36.522 | - Prerenders: 0
22:23:36.522 | - Serverless Functions: 1
22:23:36.522 | - Edge Functions: 0
22:23:36.523 | Serverless regions: Washington, D.C., USA
22:23:36.523 | Deployed outputs in 2s
22:23:36.847 | Build completed
22:23:42.432 | Uploading build cache [18.88 MB]...
22:23:43.991 | Build cache uploaded: 1.558s
```
but how come in the usage tab it shows 0 serverless functions usage? other usage stats seem to be fine (bandwidth, requests, etc.)
<img width="1728" alt="Screenshot 2023-03-14 at 21 34 50" src="https://user-images.githubusercontent.com/5616556/225143897-5e3175bc-dfea-48ed-8538-8bf7b796f74c.png">
